frontend/src/metabase/query_builder/components/view/sidebars/ChartTypeSidebar.tsx

```diff
@@ -101,10 +100,9 @@ const ChartTypeSidebar = ({
         openChartSettings(e);
       } else {
         const newQuestion = question.setDisplay(display).lockDisplay(); // prevent viz auto-selection
-
         updateQuestion(newQuestion, {
-          reload: false,
           shouldUpdateUrl: question.query().isEditable(),
+          onDisplayUpdate: visualizations.get(display).onDisplayUpdate,
```

Comment:
I'd call it earlier to update the question in ChartTypeSidebar instead of in `updateQuestion`